conans/model/build_info.py

```diff
@@ -77,8 +77,12 @@ def __init__(self):
     def _filter_paths(self, paths):
         abs_paths = [os.path.join(self.rootpath, p)
                      if not os.path.isabs(p) else p for p in paths]
+        abs_paths = [
+                    p if (p and os.path.isdir(p))
+                    else conan_v2_behavior("Warning: " + str(p) + " is not a directory")
```

Comment:
The problem I see here is that this will raise if `p` is not a directory (Conan v2). It looks like it is ok because those paths should exist (that's why you are adding them to the `cpp_info`), but if you are using the recipe in _editable_ mode those paths are not yet created (they will, hopefully, when the user build locally the library).

A different approach here, that we could consider is: if the path exists and it is not a directory, then raise... but please, add a test to check that it works for symlinks and hardlinks, some of these folders could be links.